gateway/ln-gateway/proto/gatewaylnrpc.proto

```diff
@@ -96,6 +106,17 @@ message SubscribeInterceptHtlcsResponse {
   // A unique identifier for every intercepted HTLC
   // Used to identify an intercepted HTLC through processing and settlement
   bytes intercepted_htlc_id = 11;
+
+  // Optional CircuitKey that LND needs to complete the HTLC
+  CircuitKey key = 12;
+}
+
+message CircuitKey {
+  // The id of the channel that the is part of this circuit.
+  uint64 chan_id = 1;
+
+  // The index of the incoming htlc in the incoming channel.
+  uint64 htlc_id = 2;
```

Comment:
Does this abstract well over implementations?

Reply:
It does not currently, it is specific to LND. Let me see if there's an easy refactor so that it is generalized.

Reply:
I changed the gRPC spec to just include the incoming channel id and the htlc id in the `SubscribeInterceptHtlcsResponse`. This is generic across all implementations. 

In LND we just pass these values onto LND to complete the HTLC. For our CLN implementation, we use it as the key into our `outcomes` map to lookup the channel sender needed to unblock the thread that will complete the HTLC.